elasticapm/conf/__init__.py

```diff
@@ -409,9 +440,10 @@ def reset(self):
         """
         Reset state to the original configuration
         """
-        with self._lock:
-            self._version = self._first_version
-            self._config = self._first_config
+        # We use an update rather than just overwriting the _config and _version
+        # to make sure any config values with callbacks are appropriately
+        # processed.
+        self.update(version=self._first_version, **self._first_config._values)
```

Comment:
This is definitely wrong. `_config._values` doesn't map properly to inline config (I think it would with a `.lower()` but I don't want to introduce that requirements) and in any case it won't pass the validators. Need to figured out a better way to process callbacks on a `reset()`.